// @reach/window-size/dist/window-size.esm.js
import { useRef, useState } from 'react';
import PropTypes from 'prop-types';
import { canUseDOM, useIsomorphicLayoutEffect } from '@reach/utils';

/**
 * Measure the current window dimensions.
 *
 * @see Docs   https://reacttraining.com/reach-ui/window-size
 * @see Source https://github.com/reach/reach-ui/tree/main/packages/window-size
 */

/**
 * WindowSize
 *
 * @see Docs https://reacttraining.com/reach-ui/window-size#windowsize
 * @param props
 */

var WindowSize = function WindowSize(_ref) {
  var children = _ref.children;
  var dimensions = useWindowSize();
  return children(dimensions);
};

if (process.env.NODE_ENV !== "production") {
  WindowSize.displayName = "WindowSize";
  WindowSize.propTypes = {
    children: PropTypes.func.isRequired
  };
}

/**
 * useWindowSize
 *
 * @see Docs https://reacttraining.com/reach-ui/window-size#usewindowsize
 */

function useWindowSize() {
  var _useRef = useRef(canUseDOM()),
      hasWindow = _useRef.current;

  var _useState = useState({
    width: hasWindow ? window.innerWidth : 0,
    height: hasWindow ? window.innerHeight : 0
  }),
      dimensions = _useState[0],
      setDimensions = _useState[1];

  useIsomorphicLayoutEffect(function () {
    var resize = function resize() {
      return setDimensions({
        width: window.innerWidth,
        height: window.innerHeight
      });
    };

    window.addEventListener("resize", resize);
    return function () {
      return window.removeEventListener("resize", resize);
    };
  }, []);
  return dimensions;
}

export default WindowSize;
export { WindowSize, useWindowSize };
